Dockerfile

```diff
@@ -21,34 +22,45 @@ WORKDIR /app
 RUN sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)" && \
     sudo chsh -s $(which zsh) && \
     sed -i 's/ZSH_THEME="robbyrussell"/ZSH_THEME="essembeh"/g' $HOME/.zshrc
-ENV SHELL /bin/zsh
+ENV SHELL="/bin/zsh"
 SHELL [ "/bin/zsh", "-c" ]
 
+
 # setup nvm / node / npm / pnpm
-ENV NVM_DIR ${HOME}/.nvm
+ENV NVM_DIR="${HOME}/.nvm"
 RUN mkdir -p ${NVM_DIR}
 RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | bash
-ENV NODE_VERSION 20.15.1
+ENV NODE_VERSION="20.15.1"
 RUN source ${NVM_DIR}/nvm.sh && \
     nvm install ${NODE_VERSION} && \
     nvm use ${NODE_VERSION} && \
     nvm alias default ${NODE_VERSION} && \
     rm -rf ${NVM_DIR}/.* ${NVM_DIR}/*.md ${NVM_DIR}/test ${NVM_DIR}/versions/node/v${NODE_VERSION}/*.md
-ENV PATH $NVM_DIR/versions/node/v$NODE_VERSION/bin:$PATH
+ENV PATH="$NVM_DIR/versions/node/v$NODE_VERSION/bin:$PATH"
 
 RUN npm install -g pnpm
-ENV PNPM_HOME=${HOME}/.local/share/pnpm
+ENV PNPM_HOME="${HOME}/.local/share/pnpm"
 ENV PATH="$PNPM_HOME:$PATH"
 RUN pnpm config set store-dir ${PNPM_HOME}/store && \
     pnpm setup zsh
 
-# setup python
-ENV PYENV_ROOT="${HOME}/.pyenv" PATH="${PYENV_ROOT}/shims:${PYENV_ROOT}/bin:${PATH}"
-RUN pip config set global.index-url https://mirrors.aliyun.com/pypi/simple/ && \
-    pip install wheel
+# setup uv and python
+ENV PYTHON_VERSION="3.12.1"
+ENV PATH="${HOME}/.local/uv_bin/bin/:${PATH}"
+RUN curl -LsSf https://astral.sh/uv/install.sh | env UV_INSTALL_DIR=${HOME}/.local/uv_bin bash && \
+    uv python install ${PYTHON_VERSION} 
+# ENV is a static command and cannot obtain the architecture information of the local machine.
```

Comment:
```suggestion
# We need to make python effective globally, but uv currently does not have a command to directly obtain the full path of the current python, so it can be done by manual concatenation.
# Also, there is no elegant way to get the current architecture, so we can only solve it by adding all the paths for x64 and arm64 to PATH.
```